src/ring_theory/ideal/basic.lean

```diff
@@ -490,16 +553,34 @@ namespace ideal
 /-- All ideals in a division ring are trivial. -/
 lemma eq_bot_or_top : I = ⊥ ∨ I = ⊤ :=
 begin
-  rw or_iff_not_imp_right,
-  change _ ≠ _ → _,
-  rw ideal.ne_top_iff_one,
+  rw [or_iff_not_imp_right, ← ne.def, ideal.ne_top_iff_one],
   intro h1,
   rw eq_bot_iff,
   intros r hr,
   by_cases H : r = 0, {simpa},
   simpa [H, h1] using I.mul_mem_left r⁻¹ hr,
 end
 
+@[priority 100] -- see Note [lower instance priority]
+noncomputable instance : fintype $ ideal K :=
+{ elems := {⊥, ⊤},
+  complete := λ I, by simp [eq_bot_or_top], }
+
+lemma _root_.right_ideal.eq_bot_or_top (I : right_ideal K) : I = ⊥ ∨ I = ⊤ :=
+begin
+  rw [or_iff_not_imp_right, ← ne.def, right_ideal.ne_top_iff_one],
+  intros h1,
+  rw eq_bot_iff,
+  intros r hr,
+  by_cases H : r = 0, {simpa},
+  simpa [H, h1] using I.mul_mem_right r⁻¹ hr,
+end
+
+@[priority 100] -- see Note [lower instance priority]
+noncomputable instance : fintype $ right_ideal K :=
+{ elems := {⊥, ⊤},
```

Comment:
```suggestion
instance : fintype $ right_ideal K :=
{ elems := finset.cons ⊥ {⊤} $ finset.mem_singleton.not.mpr bot_ne_top,
```

Reply:
I'm reluctant to make this change because it gives me nothing mathematically and hugely obscures the idea.

Reply:
The fact that it makes `fintype.card (ideal K) = 2` true by `rfl` seems like a strong enough argument for me. The reality is that `insert a s` is a lot more annoying to work with than `cons a s`, because you constantly have to reprove that `a` is not in `s` when expanding sums / products etc.

I agree it's unfortunate that we don't have nice notation for `cons`.

> because it gives me nothing mathematically 

Arguably this `fintype` instance gives you nothing mathematically anyway, the mathematical idea was already fully expressed in `eq_bot_or_top`. The purpose of the `fintype` instance is to be convenient, and the computable version is more so.

Reply:
I think I could argue the point but for the sake of expediency I'll capitulate.

In any case, sooner or later one I'll get Karl's new `finite` class imported into a low-level file and I'll replace a bunch of these horrid `fintype` instances with it :-)